examples/jetstream/workqueue-stream/java/Main.java

```diff
@@ -0,0 +1,171 @@
+package example;
+
+import io.nats.client.*;
+import io.nats.client.api.StreamConfiguration;
+import io.nats.client.api.ConsumerConfiguration;
+import io.nats.client.api.StreamInfo;
+import io.nats.client.api.StreamState;
+import io.nats.client.api.RetentionPolicy;
+import io.nats.client.api.AckPolicy;
+
+import java.io.IOException;
+import java.time.Duration;
+import java.nio.charset.StandardCharsets;
+
+public class Main {
+  public static void main(String[] args) {
+    String natsURL = System.getenv("NATS_URL");
+    if (natsURL == null) {
+        natsURL = "nats://127.0.0.1:4222";
+    }
+
+    // Initialize a connection to the server. The connection is AutoCloseable on exit.
+    try (Connection nc = Nats.connect(natsURL)) {
+
+      // ### Creating the stream
+      // Initialize the `JetStreamManagement` context for managing JetStream
+      // assets.
+      JetStreamManagement jsm = nc.jetStreamManagement();
+
+      // Define the stream configuration, specifying the retention policy
+      // as `WorkQueue` and create it.
+      StreamConfiguration sc = StreamConfiguration.builder()
+        .name("EVENTS")
+        .subjects("events.>")
+        .retentionPolicy(RetentionPolicy.WorkQueue)
+        .build();
+
+      jsm.addStream(sc);
+      System.out.printf("Created stream %s\n", sc.getName());
+
+      // ### Queue
+      // Create a `JetStream` context for publishing and subscribing.
+      JetStream js = nc.jetStream();
+
+      js.publish("events.us.page_loaded", null);
+      js.publish("events.eu.mouse_clicked", null);
+      js.publish("events.us.input_focused", null);
+      System.out.println("Published 3 messages");
+
+      StreamInfo info = jsm.getStreamInfo("EVENTS");
+      long msgCount = info.getStreamState().getMsgCount();
+      System.out.printf("msgs in stream: %s\n", msgCount);
+
+      // ### Adding a consumer
+      // Now let's add a consumer. It can be a *push* or *pull* consumer,
+      // however for this example we will use *pull*.
+      // The first pull consumer will be bound to all subjects in
+      // the stream. 
```

Comment:
nitpick - even though you mention it below, maybe add something like:
> since we don't specify `.filterSubject(...)` it means "all subjects in the stream"

Reply:
Done.